<!-- how to access promise -->

<!-- using = .then()and .catch() -->
 <!DOCTYPE html>
 <html lang="en">
 <head></head>
 <body></body>
 <script>
    const getPromise = ()=>{
        return new Promise((resolve,reject)=>{
            // console.log("i am promise");
            // resolve("success")
            // reject("network error")
            
        })
    }
    let promise = getPromise()
    promise.then((res)=>{
        // console.log("promise fulfilled",res);
        
    })
    promise.catch((err)=>{
        // console.log("rejected",err);
        
    })
    
    // Promise chaining...............

    function asynfun1(){
        return new Promise((resolve,reject)=>{
            setTimeout(()=>{
                console.log("data 1");
                resolve("success")
            },4000)
        })
    }
    function asynfun2(){
        return new Promise((resolve,reject)=>{
            setTimeout(()=>{
                console.log("data 2");
                resolve("success")
            },4000)
        })
    }
    console.log("fetching data1");
    let p1 = asynfun1()
    p1.then((res)=>{
        console.log("fetching data 2");
        p2= asynfun2()
        p2.catch((res)=>{})
        
    })

    // console.log()
    // let p2 =asynfun2()
    // p2.catch((res)=>{})






 </script>
 </html>